### ISSUE TITLE
Examples broken

### ISSUE THREAD

**Issue by stevenroose** [ElementsProject/rust-elements#156]
It seems that the examples are broken on master. Running this produces various compilation errors:
```
cargo test --features serde
```

**Comment by apoelstra** [comment]
Can you quote the error messages? That command runs cleanly for me and it **is** run in CI, which has also passed recently.

**Comment by stevenroose** [comment]
Hmm, came here to report this again, but probably different broken. The PSET ones are broken now. Many errors about private fields:
```
Some errors have detailed explanations: E0308, E0433, E0599, E0609, E0616.
For more information about an error, try `rustc --explain E0308`.
error[E0616]: field `inputs` of struct `elements::pset::PartiallySignedTransaction` is private
   --> examples/pset.rs:298:10
    |
298 |     pset.inputs[0].partial_sigs.clear();
    |          ^^^^^^ private field
    |
help: a method `inputs` also exists, call it with parentheses
    |
298 |     pset.inputs()[0].partial_sigs.clear();
    |                ++

error[E0616]: field `inputs` of struct `elements::pset::PartiallySignedTransaction` is private
   --> examples/pset.rs:299:10
    |
299 |     pset.inputs[0].final_script_witness = Some(vec![
    |          ^^^^^^ private field
```

**Comment by sanket1729** [comment]
Hmm, this should definitely be added to the CI.

**Comment by delta1** [comment]
@apoelstra I think they are run as part of [`test.sh`](https://github.com/ElementsProject/rust-elements/blob/master/contrib/test.sh#L24) ? or does your local CI not use that script

**Comment by apoelstra** [comment]
@delta1 no, my local CI uses nix so that I can reuse all the dependencies' builds across runs. `cargo` rebuilds the world every time.

**Comment by delta1** [comment]
cool.. maybe a `postCheck` that calls `run-parts` ?

**Comment by apoelstra** [comment]
I guess you mean using `run-parts` to call the contrib scripts? No, because the contrib scripts call `cargo` which will rebuild the world and take 10x as long.

I think to run examples I would need https://github.com/kolloch/crate2nix/issues/284 -- it looks like there is a draft PR open so I can crib code from that.

**Comment by apoelstra** [comment]
It looks like the `target/debug/examples` directory isn't created. I think I would need to modify crate2nix.
